packages/@react-spectrum/combobox/src/MobileComboBox.tsx

```diff
@@ -55,20 +56,39 @@ export const MobileComboBox = React.forwardRef(function MobileComboBox<T extends
     isQuiet,
     isDisabled,
     validationState,
-    isReadOnly
+    isReadOnly,
+    menuTrigger = 'input'
   } = props;
 
   let {contains} = useFilter({sensitivity: 'base'});
   let state = useComboBoxState({
     ...props,
     defaultFilter: contains,
     allowsEmptyCollection: true,
-    shouldCloseOnBlur: false
+    shouldCloseOnBlur: true
```

Comment:
This fixes the following bug: "When you blur from the field in Android, the input value doesn't reset to the selected key's text". However, runs counter to the behavior setup in this pull: https://github.com/adobe/react-spectrum/pull/1176 and this commit specifically: https://github.com/adobe/react-spectrum/commit/39be0d7ab9bf1391ef4560281df1a0ce43fde85b

If we want to preserve the behavior where the combobox value isn't committed upon closing the tray but have the input value reset when blurring from the mobile combobox button, we'll need to expose `commitSelection` or something from `useComboBoxState` and use it in the mobile combobox button blur perhaps.

Reply:
e9d81f9565f173329e04cbe1aff280de2fd4a164 makes it so the input value doesn't reset when the tray input gets blurred even though `shouldCloseOnBlur` is true (early returns `onBlur` so `setFocused` from useComboBoxState doesn't get called), but then allows blurring the combobox button to reset the input value because `shouldCloseOnBlur` is true

Reply:
I think it's weird to keep it until the outer button is blurred. Would it make more sense to have it reset when the tray closes? Looks like we already call `commit` on blur, but I guess it does nothing in this case because nothing is focused? Maybe there should be an `else` case there that calls `resetInputValue`?

Reply:
Hrm, lemme dig, but I think we won't be able to add a `else` that calls `resetInputValue` (aka `resetInputValue` if `commit` is called and there isn't a `focusedKey`) since it would cause the following flow to break:
1. User types in the field and the menu opens
2. No item is focused and user hits enter key
3. Nothing should happen but with this change the input value would get wiped.

Reply:
Hmm. Is that weird that nothing happens if you don't have an item focused? Should it still close the menu and reset? Though I guess on mobile we wouldn't want it to close when pressing enter? We could override it to just blur the input and hide the keyboard I guess...

Alternatively, we could expose another method on the state that resets the value and call that when the tray closes.

Either way, it feels weird right now that this fix is reliant on changes in 3 seemingly unrelated places. It would be nice to make it a bit more explicit.

Reply:
So this would depend on what we would want the non-mobile case to do as well. If a user types in the field and hits enter, do we want it to close the menu and reset the text? I _think_ we discussed this as a team previously and decided that it should do nothing since it matched v2 combobox.

I'm leaning towards returning `resetInputValue` in state, lemme change that

Reply:
Another interesting case is that hitting the escape key doesn't call the Tray's onClose since useComboBox handles closing the tray on escape. I'll have mobile combobox change the escape key behavior to do `commit -> reset -> close` flow as well

Reply:
Ok, @dannify and I discussed and I think we came to the following conclusions:

* Pressing Enter should always do something. If you haven't focused any items, then it should reset the input value and close the menu. I think that should avoid the need to expose state.resetInputValue. On mobile, we should override this keyboard listener and only hide the software keyboard rather than close the tray (i.e. move focus to the listbox).
* Pressing Escape should also reset and close the menu/tray.

What do you think? Also, I'm sorry. 😉

Reply:
Sounds good, I'll give it a shot. One clarification for the following case: MobileComboBox w/ allowsCustomValue. I presume that pressing escape should only reset the input value if there is a selectedKey? IMO if there was a custom value previously entered and the user opens the tray -> modifies the input -> hits escape, the new value should replace the old value (this mirrors what happens in desktop combobox at the moment).

Another clarification: should pressing Escape really reset the input value for desktop combobox? IMO what we have for the desktop combobox right now is fine i.e. hitting escape closes the menu only, blurring the field resets the input.

Reply:
I think of Escape as a "cancel" action rather than only "close". So IMO it should behave exactly like onBlur.

To answer your questions:

1. Yes, it should do whatever `onBlur` does now. I think for `allowsCustomValue`, it would only close the menu/tray and not try to reset (i.e. `commitCustomValue`) otherwise reset (`commitSelection`).
2. I think so? Again, I see it more as a cancel.

I guess this should be exposed as a method on the state like `state.revert()`. So we'd have `state.commit` and `state.revert`, which seems to make sense to me.

Reply:
Regarding moving the focus to the listbox when Enter is pressed on mobile, do we really want to do that or was that just a suggestion on how to close the software keyboard? Feels like that breaks the virtual focus setup that we have since we have all of our keydown handlers on the input itself.

I can close the virtual keyboard by setting readonly on to the field -> blurring the field -> removing the readonly if that sounds fine to you